lib/handlebars/runtime.js

```diff
@@ -136,6 +136,9 @@ export function template(templateSpec, env) {
       }
 
       if (resultIsAllowed(result, container.protoAccessControl, propertyName)) {
+        if (typeof result === 'function') {
+          return parent[propertyName]();
```

Comment:
Why did you use `parent[propertyName]()` instead of `result()`? Any specific reason?

Reply:
It doesn't work with `result()`, complains that:

```
TypeError: String.prototype.trim called on null or undefined
```

I'm not 100% sure why, but as I understand `result` loses `this` context.